common/static/common/js/discussion/utils.js

```diff
@@ -386,51 +413,55 @@
                         return $3;
                     });
                 } else {
-                    processedText += text;
+                    processedText += tempText;
                     break;
                 }
             }
             /* jshint +W083 */
-            text = processedText;
-            text = text.replace(new RegExp(ESCAPED_DOLLAR, 'g'), '\\$');
-            text = text.replace(/\\\\\\\\/g, ESCAPED_BACKSLASH);
-            text = text.replace(/\\begin\{([a-z]*\*?)\}([\s\S]*?)\\end\{\1\}/img, function($0, $1, $2) {
+            tempText = processedText;
+            tempText = tempText.replace(new RegExp(ESCAPED_DOLLAR, 'g'), '\\$');
+            tempText = tempText.replace(/\\\\\\\\/g, ESCAPED_BACKSLASH);
+            tempText = tempText.replace(/\\begin\{([a-z]*\*?)\}([\s\S]*?)\\end\{\1\}/img, function($0, $1, $2) {
                 return processor(("\\begin{" + $1 + "}") + $2 + ("\\end{" + $1 + "}"));
```

Comment:
Same issue here.  My main issue with processor is that it seems like it should take and return and HTML Snippet, or take and return a String with HTML, but doesn't seem right to take one and return the other.  I'm just repeating myself now.